mlflow/store/tracking/file_store.py

```diff
@@ -931,3 +970,91 @@ def record_logged_model(self, run_id, mlflow_model):
             self._set_run_tag(run_info, tag)
         except Exception as e:
             raise MlflowException(e, INTERNAL_ERROR)
+
+
+def _like_pattern_to_regex(pattern):
+    """
+    Convert a pattern string for LIKE and ILIKE to the equivalent regular expression.
+    """
+    return "^" + pattern.replace("%", ".*") + "$"
+
+
+def _create_matcher(comparator, value):
+    if comparator == "=":
+        return value.__eq__
+    elif comparator == "!=":
+        return value.__ne__
+    elif comparator in ("LIKE", "ILIKE"):
+        flag = re.IGNORECASE if comparator == "ILIKE" else 0
+        regex = re.compile(_like_pattern_to_regex(value), flag)
+        return regex.match
+    raise MlflowException.invalid_parameter_value(f"Invalid comparator: {comparator}")
```

Comment:
Perhaps we want to place this in `SearchUtils`? The existing `FileStore.search_runs()` API just calls out to `SearchUtils`. Would be nice to encapsulate the code in a consistent location.

Reply:
Removed this function while addressing https://github.com/mlflow/mlflow/pull/6227#discussion_r919155658.